vpc/service/db/test/test_util.go

```diff
@@ -0,0 +1,230 @@
+package db_test
+
+import (
+	"context"
+	"crypto/rand"
+	"database/sql"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/Netflix/titus-executor/logger"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/lib/pq"
+)
+
+type PostgresContainer struct {
+	id       string // docker container id
+	password string // psql password
+	port     string // psql port
+}
+
+// Start a postgres container locally for testing.
+func StartPostgresContainer(ctx context.Context) (*PostgresContainer, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		return nil, err
+	}
+	defer cli.Close()
+	const IMAGE string = "postgres"
+	r, err := cli.ImagePull(ctx, IMAGE, types.ImagePullOptions{})
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+	_, err = io.Copy(io.Discard, r)
+	if err != nil {
+		return nil, err
+	}
+
+	password, err := randomPassword()
+	if err != nil {
+		return nil, err
+	}
+
+	port, err := randomPort()
+	if err != nil {
+		return nil, err
+	}
+
+	createResp, err := cli.ContainerCreate(ctx, &container.Config{
+		Image: IMAGE,
+		Env: []string{
+			"POSTGRES_DB=pgtest",
+			"POSTGRES_PASSWORD=" + password,
+			"POSTGRES_USER=pgtest",
+		},
+		Healthcheck: &container.HealthConfig{
+			Test:     []string{"CMD-SHELL", "pg_isready -U pgtest"},
+			Interval: time.Second,
+			Timeout:  time.Second,
+			Retries:  10,
+		},
+	}, &container.HostConfig{
+		PortBindings: nat.PortMap{
+			"5432/tcp": []nat.PortBinding{
+				{HostPort: port},
+			},
+		},
+	}, nil, "pgtest")
+	if err != nil {
+		return nil, err
+	}
+	container := &PostgresContainer{
+		id:       createResp.ID,
+		password: password,
+		port:     port,
+	}
+	err = cli.ContainerStart(ctx, createResp.ID, types.ContainerStartOptions{})
+	if err != nil {
+		_ = cli.ContainerRemove(ctx, container.id, types.ContainerRemoveOptions{})
+		return nil, err
+	}
+
+	done := make(chan bool)
+	go func() {
+		for {
+			resp, err := cli.ContainerInspect(ctx, createResp.ID)
+			if err == nil && resp.State.Health.Status == types.Healthy {
+				done <- true
+				break
+			}
+			time.Sleep(500 * time.Millisecond)
+		}
+	}()
+	select {
+	case <-done:
+		logger.G(ctx).Infof("started container %s on port %s", container.id, container.port)
+		return container, nil
+	case <-time.After(time.Minute):
+		err = container.Shutdown(ctx)
+		if err != nil {
+			logger.G(ctx).Errorf("failed to clean up container: %s", err)
+		}
+		return nil, errors.New("container health check failed")
+	}
+}
+
+// Connect to the test DB and set up table schema using the DB migrate tool
```

Comment:
Was the intent to call `err = db.MigrateTo(ctx, testDb, 40, false)` here? (according to the comment)

Reply:
Oh I tried putting it there in the first place. Somehow `make test-local` works on my machine but failed remotely when I do that. Updated the comment.